stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/models/FilterObject.kt

```diff
@@ -1,33 +1,37 @@
 package io.getstream.chat.android.client.api.models
 
-internal sealed class FilterObject
+public sealed class FilterObject
 
-internal data class AndFilterObject(val filterObjects: List<FilterObject>) : FilterObject()
+public data class AndFilterObject(val filterObjects: Set<FilterObject>) : FilterObject()
```

Comment:
When do we want clients to create these by calling the constructors directly instead of going through the `Filters` helper? Do they really have to be public?

Reply:
They are needed on Offline-SDK, maybe we can keep internal constructor

Reply:
`internal` constructors would be great, we could also just make it all `@InternalStreamChatApi` as well